examples/capi-quick-start/docker-cluster-class.yaml

```
@@ -4,52 +4,30 @@
 apiVersion: cluster.x-k8s.io/v1beta1
 kind: ClusterClass
 metadata:
-  name: capd-quick-start
+  labels:
+    cluster.x-k8s.io/provider: docker
+  name: docker-quick-start
 spec:
   controlPlane:
     machineInfrastructure:
       ref:
         apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
         kind: DockerMachineTemplate
-        name: capd-quick-start-control-plane
+        name: aws-quick-start-control-plane
```

I'm still looking how this is being set, `kustomize` reverts it back 🤔 
```suggestion
        name: docker-quick-start-control-plane
```

---

kustomize is getting confused here because both base ClusterClasses have a ref `AWSMachineTemplate` named `quick-start-control-plane`. Being more specific adding `apiVersion` to `nameReference` didn't help. Surprisingly this doesn't happen for the other refs.
Do we need separate `kustomize` executions here for each provider?

---

I thought that you could use `nameReference` configs in each sub-kustomization but that didn't work. I have refactored the kustomization to do this via patches instead. It's not as convenient, but works.